blog-posts/gatsby-ssr-caching.md

```diff
@@ -0,0 +1,95 @@
+
+With those 3 directives, you can make your page as fast as with SSG or DSG. If the values are set correctly, only the 
+very first user after a deployment will have to wait for the rendering. Everyone else will see a lightning fast page.
+
+> This header can also be used on any other SSR framework and CDN. It's not limited to Gatsby or Gatsby Cloud.
+
+## Enable Caching
+
+Now let’s dive into the code and set the `Cache-Control` header:
+
+```jsx
+// pages/ssr.jsx
+export default function Cache() {
+  return (
+      <h1>SSR: Server Side Rendering</h1>
+  )
+}
+
+export async function getServerData() {
+  return {
+    headers: {
+      'Cache-Control': 'public, max-age=10, s-maxage=60, stale-while-revalidate=240',
+    }
+  }
+}
+```
+
+Gatsby SSR pages export the `getServerData()`  function, which you can use to do all kinds of stuff with it (like 
+fetching external data). Besides the props that can be injected into the page itself, this function returns a `headers` 
+object. This object is used to set the `Cache-Control` header.
+
+In our case, we set the browser cache to 10 seconds and the CDN cache to 1 minute. After 1 minute, the CDN will 
+serve stale content for a maximum of 4 additional minutes. If there is one request within these 4 minutes, a 
+new page will be rendered in the background while the old page gets served. After the new page is rendered, it will replace 
+the old one on the CDN and the caching time starts again. Let’s check the network tab to see if it works:
+
+![Chrome network tab with proper caching header. Everything works as expected.](images/gatsby-ssr-caching-network-tab.png)
+
+You can see: The very first page took 5 seconds to load (for this example, I added an artificial delay of 5 seconds to make the server rendered 
+page more obvious). All other requests were served from CDN within 30ms. That’s super fast! Also, after the stale page 
+was served, the page was re-rendered in the background and replaced. The request after the stale one was super fast 
+again but served the new page.
+
+## Conclusion
+
+SSR with proper CDN caching is very powerful and can be used for a lot of scenarios, especially where you hit the 
+limits of SSG and DSG. The integration within Gatsby and Gatsby Cloud is seamless, you can introduce it on a per-page basis.
+
+Some things to keep in mind:
+
+- SSR functionality is quite new in Gatsby and may not be as mature as in other frameworks.
+- Same goes for Gatsby Cloud, it’s quite new and may not be as mature as other hosting services.
+- SSR officially only works with Gatsby Cloud for now. You cannot host it somewhere else. There are third-party plugins though for [Fastify](https://github.com/gatsby-uc/plugins/tree/main/packages/gatsby-plugin-fastify) (self-hosting) or [Netlify](https://github.com/netlify/netlify-plugin-gatsby) (SAAS).
+- `gatsby-plugin-image` doesn’t work with SSR.
```

Comment:
new facts in conclusion  are never good. I think those are good for an extra section?